src/version.c

```diff
@@ -63,6 +68,7 @@ int get_latest_version(void)
 	free(path);
 	free(url);
 	free(tmp_version.version);
+	cached_version = ret;
```

Comment:
It might be better to set `cached_version` outside the error path (i.e. immediately before the `out` label).